src/corppa/poetry_detection/annotation/recipe.py

```diff
@@ -234,12 +243,106 @@ def annotate_page_text(
         "config": config,
     }
 
+    # remove fetched image data before saving to the database
     if fetch_media:
-        components["before_db"] = lambda x: remove_images(x, image_prefix=image_prefix)
+        components["before_db"] = lambda x: remove_image_data(
+            x, image_prefix=image_prefix
+        )
 
     return components
 
 
+def get_session_name(example: TaskType, suffix: Optional[str] = None) -> str:
+    """
+    Extract session name from task examplke. Session ids have the following form:
+        [session id] = [db id]-[session name]
+    Assumes that session names do not contain dashes.
+    """
+    session_id = example[SESSION_ID_ATTR]
+    session_name = session_id.rsplit("-", maxsplit=1)[-1]
+    if suffix is not None:
+        session_name = f"{session_name}-{suffix}"
+    return session_name
+
+
+def add_label_prefix(label: str, prefix: str) -> str:
+    return f"{prefix}: {label}"
+
+
+def remove_label_prefix(label: str) -> str:
+    return label.rsplit(": ", maxsplit=1)[-1]
+
+
+def add_session_prefix(
+    example: TaskType, session_sfx: Optional[str] = None
+) -> TaskType:
+    """
+    Add session name as prefix to text span labels
+
+    label --> "[session]: label"
+    ex. "POETRY" --> "alice: POETRY"
+
+    Calls: `get_sesssion_name` and `add_label_prefix`
+    """
+    session_name = get_session_name(example, suffix=session_sfx)
+    for span in example.get("spans", []):
+        span["label"] = add_label_prefix(span["label"], session_name)
+    return example
+
+
+def remove_session_prefix(example: TaskType) -> TaskType:
+    """
+    Remove added session prefix from text span labels
+
+    Calls: `remove_label_prefix`
+    """
+    for span in example.get("spans", []):
+        span["label"] = remove_label_prefix(span["label"])
+    return example
+
+
+def has_span_overlap(example: TaskType, strip_label_pfx: bool = True) -> bool:
+    """
+    Check if example has overlapping (text) span annotations
+
+    Calls: `remove_label_prefix` (optionally)
+    """
+    if "spans" not in example:
+        return False
+    label_coverage = {}  # label (str) --> coverage (intspan)
+    for span in example["spans"]:
+        # Get label
+        label = span["label"]
+        if strip_label_pfx:
+            label = remove_label_prefix(label)
+        # Represent span's coverage as an intspan
+        # Note: intspans are closed intervals, while span character ranges are half-open
+        span_coverage = intspan.from_range(span["start"], span["end"] - 1)
+        # Check if label's intspan is disjoint from
```

Comment:
is this comment in complete and/or in the wrong place?

Reply:
Good catch, looks like the comment got cut-off. It should be
```suggestion
        # Check if span's coverage is disjoint from label's existing coverage.
```